### PR TITLE
Add the ability to limit the UI to specific process groups.

### PR DESCRIPTION
**Pull request by Sander-Toonen**
I have multiple sites running on the same server. Each of these sites have processes managed by Supervisor. These processes are kept in a process group per site. In the admin area of a specific site, I'd like to show the state of the processes. However, as it is now all the processes are shown, including those of other sites. 

This PR will add the ability to limit the UI to showing processes in specific process groups.

Note: This PR is not ready to be merged! Instead I'd like to hear your opinion whether this approach would be a good way to solve the problem.

Depends on PR: https://github.com/yzalis/Supervisor/pull/2

### CODE REVIEW

**Comment by Sander-Toonen** [comment]
@blaugueux  Is there anything I can do to move this PR forwards?

File: Manager/LimitedSupervisor.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * The constructor
+     *
+     * @param string $name
+     * @param string $ipAddress The server ip adress
+     * @param string $username Default set to null
+     * @param string $password Default set to null
+     * @param integer $port Default set to null
+     * @param array $limitToGroups Groups to limit this supervisor to.
+     */
+    public function __construct($name, $ipAddress, $username = null, $password = null, $port = null, $limitToGroups = array())
```
Comment (blaugueux) [inline Manager/LimitedSupervisor.php]: Please use ```[]``` instead of ```array()```

File: Manager/LimitedSupervisor.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    /**
+     * getProcesses
+     *
+     * @param array $limitToGroups Only show processes in these process groups.
+     *
+     * @return Process[]
+     */
+    public function getProcesses($limitToGroups = array())
```
Comment (blaugueux) [inline Manager/LimitedSupervisor.php]: Please use ```[]``` instead of ```array()```

File: Manager/LimitedSupervisor.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * LimitedSupervisor
+ */
+class LimitedSupervisor extends Supervisor
+{
+    /**
+     * @var array
+     */
+    protected $limitToGroups;
```
Comment (blaugueux) [inline Manager/LimitedSupervisor.php]: I'm not a huge fan of this variable name and the class name.

IMHO ```Limited``` is too generic and could be more specific. ```limitToGroups``` should be ```groups```.

File: Manager/LimitedSupervisor.php (hunk 1 above)
Comment (Sander-Toonen) [inline Manager/LimitedSupervisor.php]: With all my pleasure. I love the []-syntax :)

File: Manager/LimitedSupervisor.php (hunk 3 above)
Comment (Sander-Toonen) [inline Manager/LimitedSupervisor.php]: Thank you for your feedback. I'll change ``limitToGroups`` to ``groups``. Regarding the classname, what about ``GroupFilteredSupervisor``?

Comment (blaugueux) [inline Manager/LimitedSupervisor.php]: ```GroupRestrictedSupervisor```?

**Comment by Sander-Toonen** [comment]
Note that a minor version bump of ``yzalis/Supervisor`` after merging https://github.com/yzalis/Supervisor/pull/2 and an increase of the version constraint in composer.json are needed before it can be merged.

**Comment by blaugueux** [comment]
Can you squash your commits please.

**Comment by Sander-Toonen** [comment]
Done

**Comment by Sander-Toonen** [comment]
@blaugueux Anything more before it is good to go?

**Comment by blaugueux** [comment]
@Sander-Toonen Sorry i'm so busy. I'll have time next weekend to manage this PR.

**Comment by Sander-Toonen** [comment]
@blaugueux Ping! ;)

**Comment by blaugueux** [comment]
Thanks for your work @Sander-Toonen 👍

**Comment by Sander-Toonen** [comment]
Thank you for this bundle!

**Pull request merged**